kie-kogito-docs-554: Guide to help DB Admins to understand Serverless Workflows persistence (#555)
apache/incubator-kie-kogito-docs: serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/persistence/persistence-core-concepts.adoc

= Core Concepts
:compat-mode!:
// Metadata:
:description: Running Quarkus Workflow Application using PostgresSQL
:keywords: kogito, workflow, quarkus, serverless, quarkus-cli, persistence, postgresql

> The SonataFlow workflows runtime persistence is the mechanism to ensure that your workflow instances remain consistent in the context of an error or a runtime restart. For example, a pod restart, a programmed maintenance shutdown, etc.

Reviewer: ```suggestion
The {product_name} workflow runtime persistence is the mechanism to ensure that your workflow instances remain consistent in the context of an error or a runtime restart. For example, a pod restart, a programmed maintenance shutdown, etc.
```

> The SonataFlow workflows runtime persistence is the mechanism to ensure that your workflow instances remains consistent in the context of an error or a runtime restart. For example, a pod restart, a programmed maintenance shutdown, etc.

Reviewer 2: ```suggestion
The SonataFlow workflows runtime persistence is the mechanism to ensure that your workflow instances remain consistent in the context of an error or a runtime restart. For example, a pod restart, a programmed maintenance shutdown, etc.
```

> Every workflow instance requires some status information and data to execute, this information is automatically managed by the workflows runtime, and is persisted at different moments of the workflow execution.

Reviewer 2: ```suggestion
Every workflow instance requires some status information and data to execute, this information is automatically managed by the workflow's runtime, and is persisted at different moments of the workflow execution.
```

> Every workflow instance requires some status information and data to execute, this information is automatically managed by the workflow's runtime, and is persisted at different moments of the workflow execution.

Reviewer: ```suggestion
Every workflow instance requires some status information and data to execute, this information is automatically managed by the workflow's runtime and is persisted at different moments of the workflow execution.
```

> For example, when a workflow instance reaches a state that needs to wait for an event, the engine takes a snapshot of the most relevant information, stores it in the database, and pause that instance execution.

Reviewer 2: ```suggestion
For example, when a workflow instance reaches a state that needs to wait for an event, the engine takes a snapshot of the most relevant information, stores it in the database, and pauses that instance execution.
```

> In this way, resources like memory are released, and can be used by other executing instances.

Reviewer: ```suggestion
In this way, resources like memory are released and can be used by other executing instances.
```
